src/NuGetForUnity/Editor/Ui/NugetWindow.cs

```diff
@@ -934,44 +967,71 @@ private void DrawPackage(INugetPackage package, GUIStyle packageStyle, GUIStyle
-                    GUILayout.Label($"Version {package.PackageVersion.FullVersion}");
+                    GUILayout.Label($"Installed Version: {installed.PackageVersion.FullVersion}");
                 }
-                else
+
+                // Show the version selection dropdown only on Updates tab OR on Online tab if the package is not installed and not already in Unity
+                if (currentTab == NugetWindowTab.UpdatesTab ||
+                    currentTab == NugetWindowTab.OnlineTab && installed == null && !UnityPreImportedLibraryResolver.IsAlreadyImportedInEngine(
+                        package,
+                        false))
                 {
-                    if (!versionDropdownDataPerPackage.TryGetValue(package.Id, out var versionDropdownData))
+                    if (package.Versions.Count <= 1)
                     {
-                        EditorStyles.popup.CalcMinMaxWidth(
-                            new GUIContent(
-                                package.Versions.Select(version => version.FullVersion).OrderByDescending(version => version.Length).First()),
-                            out _,
-                            out var maxWidth);
-                        versionDropdownData = new VersionDropdownData
-                        {
-                            SortedVersions = package.Versions.OrderByDescending(version => version).ToList(), CalculatedMaxWith = maxWidth + 5,
-                        };
-                        versionDropdownData.DropdownOptions = versionDropdownData.SortedVersions.Select(version => version.FullVersion).ToArray();
-                        versionDropdownData.SelectedIndex = versionDropdownData.SortedVersions.IndexOf(package.PackageVersion);
-                        versionDropdownDataPerPackage.Add(package.Id, versionDropdownData);
+                        GUILayout.Label($"Version {package.PackageVersion.FullVersion}");
                     }
+                    else
+                    {
+                        if (!versionDropdownDataPerPackage.TryGetValue(package.Id, out var versionDropdownData))
+                        {
+                            EditorStyles.popup.CalcMinMaxWidth(
+                                new GUIContent(
+                                    package.Versions.Select(version => version.FullVersion).OrderByDescending(version => version.Length).First()),
+                                out _,
+                                out var maxWidth);
+
+                            if (installed != null)
+                            {
+                                // we're showing only lower or only higher versions, depending on the 'Show Downgrades' checkbox.
+                                // This can only happen in Updates tab.
+                                if (showDowngrades)
+                                {
+                                    package.Versions.RemoveAll(version => version >= installed.PackageVersion);
```

Comment:
I don't think it is a good idea to manipulate the information of the package directly I would create a new list containing only the package version's that should be visible in the UI. Imagine a package that depends on a version you removed from the "UI" so it can't be installed as you removed the information that the version exists.
I think this should also remove the need to re-download all packages by calling `UpdateUpdatePackages` when the user toggles the `showDowngrades`.

Reply:
I will fix this by simply filtering what goes into `DropdownOptions` better, and on toggling I will just clear `versionDropdownDataPerPackage`.